### ISSUE TITLE
CSE variable not initialized in code-generation (Jacobian)

### ISSUE THREAD

**Issue by JothanChanes** [OpenModelica/OpenModelica#7425]
Hi,

OpenModelica 1.17 with new front-end. The following MWE generates a compiler error because of an undeclared variable:

   >model tanktest
   >import Modelica.Media.IdealGases.SingleGases.H2;
   >import Modelica.Fluid.Valves.ValveCompressible;
   >import Modelica.Fluid.Vessels.ClosedVolume;
   >import Modelica.Fluid.System;
  >inner System system;
  >ClosedVolume buffer( redeclare package Medium = H2, T_start = 300,V = 3,nPorts = 1, p_start = 3e5, use_portsData = false);
  >ValveCompressible v1( redeclare package Medium = H2, opening = 1, Av = 1e-3, CvData = Modelica.Fluid.Types.CvTypes.Av, m_flow_nominal=1, dp_nominal = 1e5);
  >ValveCompressible v2(redeclare package Medium = H2,Av = 1e-4, opening = 1, CvData = Modelica.Fluid.Types.CvTypes.Av, m_flow_nominal=1, dp_nominal = 1e5);
  >ClosedVolume tank( redeclare package Medium = H2,T_start = 300, V = 1,nPorts = 1, p_start = 1e5, use_portsData = false);
>equation
  >connect(buffer.ports[1], v1.port_a);
  >connect(v1.port_b, v2.port_a);
  >connect(v2.port_b, tank.ports[1]);
  >annotation(
   > uses(Modelica(version = "3.2.3")));
>end tanktest;

The generated code is (tanktest_12jac.c)

>void tanktest_eqFunction_148(DATA *data, threadData_t *threadData, ANALYTIC_JACOBIAN *jacobian, ANALYTIC_JACOBIAN *parentJacobian)
>{
>  TRACE_PUSH
>  const int clockIndex = 0;
>  const int equationIndexes[2] = {1,148};
>  jacobian->tmpVars[1] /* tank.mb_flow.$pDERLSJac69.dummyVarLSJac69 JACOBIAN_DIFF_VAR */ = (_omcQ_24cse1) * ((data->simulationInfo->realParameter[66] /* v2.m_flow_nominal PARAM */) * (DIVISION(jacobian->seedVars[0] /* >v2.port_a.p.SeedLSJac69 SEED_VAR */,data->simulationInfo->realParameter[61] /* v2.dp_nominal PARAM */,"v2.dp_nominal")));
>  TRACE_POP
>}

The error is

`tanktest_12jac.c:31:259: error: use of undeclared identifier '_omcQ_24cse2'`

Thanks for having a look at it. Let me know if additional information is required.

**Comment by casella** [comment]
@JothanChanes, thanks for reporting. 

@kabdelhak, I tried to inspect the model with the debugger, when I double-click on `$pDERLSJac3`, which is apparently the culprit, no information is displayed. I guess dummy derivatives/ASSC are involved. Would you mind having a look?

**Comment by bilderbuchi** [comment]
I just checked with a recent nightly (OpenModelica v1.18.0-dev-336-gcb76b3bfca (64-bit)), and this is still present (probably not surprisingly).

**Comment by sturmk** [comment]
I am running into a very similar issue while using the ThermoPower library. Also connecting two tanks, and with a ValveLiq and Flow1DFV between the tanks.
Is there an update on this issue?

**Comment by casella** [comment]
@kabdelhak could you please check?

**Comment by casella** [comment]
@kabdelhak can we get through this issue? Thanks!

**Comment by kabdelhak** [comment]
> [@kabdelhak](https://github.com/kabdelhak) can we get through this issue? Thanks!

seems like my partial fix from last year did not survive some sanity checks, will try to finalize the PR

**Comment by casella** [comment]
Still not there 😢

**Comment by kabdelhak** [comment]
> Still not there 😢

No, i took a step back and made a new PR. This portion of the old backend is very fragile and i will try to do baby steps in the correct direction.
